Source/WebCore/Configurations/WebCore.xcconfig

```diff
@@ -75,6 +75,25 @@ WK_ANGLE_LDFLAGS = -weak-lANGLE-shared;
 
 WK_WEBGPU_LDFLAGS = -weak_framework WebGPU;
 
+// Vision.framework is not present in the watchOS SDK.
+// It's present in the macOS SDK, but isn't present in the macOS base system at runtime.
+// So we have to weak-link it on macOS.
+// And we have to weak-link it everywhere else, because Vision depends on UIKit, which depends on WebCore.
```

Comment:
I think weak-linking still introduces a circular dependency?

Reply:
(Marking PR as a draft while figuring this out)

Reply:
You're right! Ideally we'd be able to use "upward linking" but we can't because WebCore doesn't support building a text-based API. 😞 Weak linking does still have the circular dependency; soft linking is the only workable solution.